sci-libs/cholmod/cholmod-2.1.2-r1.ebuild

```diff
@@ -14,24 +14,32 @@ SLOT="0"
 KEYWORDS="~alpha amd64 ~arm arm64 hppa ~ia64 ~mips ppc ppc64 sparc x86 ~amd64-linux ~x86-linux ~x86-macos"
 IUSE="cuda doc lapack metis minimal static-libs"
 
+# doesn't build against current version of sci-libs/amd, sci-libs/colamd
+# <sci-libs/parmetis-4 is no longer in the tree
 RDEPEND="
-	>=sci-libs/amd-2.3
-	>=sci-libs/colamd-2.8
+	=sci-libs/amd-2.3*
+	=sci-libs/colamd-2.8*
 	cuda? ( x11-drivers/nvidia-drivers dev-util/nvidia-cuda-toolkit )
 	lapack? ( virtual/lapack )
 	metis? (
-		>=sci-libs/camd-2.3
+		=sci-libs/camd-2.3*
 		>=sci-libs/ccolamd-2.8
-		|| ( <sci-libs/metis-5 <sci-libs/parmetis-4 ) )"
+		<sci-libs/metis-5 )"
 
 DEPEND="${RDEPEND}
 	virtual/pkgconfig
 	doc? ( virtual/latex-base )"
 
+PATCHES=(
+	"${FILESDIR}/${P}-0001-Gentoo-specific-fix-self-linking-of-files.patch"
+)
+
 src_prepare() {
 	# bug #399483 does not build with parmetis-3.2
 	has_version "=sci-libs/parmetis-3.2*" && \
 		epatch "${FILESDIR}"/${PN}-1.7.4-parmetis32.patch
```

Comment:
Don't use `epatch` in >=EAPI-6. Either replace with `eapply`, or add it to `PATCHES` in that condition.

But then it appears this whole block can be dropped now anyway, as <sci-libs/parmetis-4 is gone?

Reply:
Was thinking, I already changed this. I'm gonna fix it.